src/plugins/data/common/search/expressions/eql.ts

```diff
@@ -0,0 +1,171 @@
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { i18n } from '@kbn/i18n';
+import { buildEsQuery } from '@kbn/es-query';
+import { ExpressionFunctionDefinition } from 'src/plugins/expressions/common';
+
+import { EqlSearchRequest } from '@elastic/elasticsearch/lib/api/typesWithBodyKey';
+import { RequestStatistics, RequestAdapter } from '../../../../inspector/common';
+import {
+  ISearchGeneric,
+  KibanaContext,
+  EqlSearchStrategyResponse,
+  EQL_SEARCH_STRATEGY,
+  EqlSearchStrategyRequest,
+} from '..';
+import { getEsQueryConfig } from '../../es_query';
+import { UiSettingsCommon } from '../..';
+import { EqlRawResponse } from './eql_raw_response';
+
+const name = 'eql';
+
+type Input = KibanaContext | null;
+type Output = Promise<EqlRawResponse>;
+
+interface Arguments {
+  query: string;
+  index: string;
+  size: number;
+  field: string[];
+}
+
+export type EqlExpressionFunctionDefinition = ExpressionFunctionDefinition<
+  typeof name,
+  Input,
+  Arguments,
+  Output
+>;
+
+interface EqlStartDependencies {
+  search: ISearchGeneric;
+  uiSettingsClient: UiSettingsCommon;
+}
+
+export const getEqlFn = ({
+  getStartDependencies,
+}: {
+  getStartDependencies: (getKibanaRequest: any) => Promise<EqlStartDependencies>;
+}) => {
+  const eql: EqlExpressionFunctionDefinition = {
+    name,
+    type: 'eql_raw_response',
+    inputTypes: ['kibana_context', 'null'],
+    help: i18n.translate('data.search.eql.help', {
+      defaultMessage: 'Run Elasticsearch request',
+    }),
+    args: {
+      query: {
+        types: ['string'],
+        aliases: ['_', 'q', 'query'],
+        help: i18n.translate('data.search.eql.q.help', {
+          defaultMessage: 'Query DSL',
+        }),
+        required: true,
+      },
+      index: {
+        types: ['string'],
+        help: i18n.translate('data.search.eql.index.help', {
+          defaultMessage: 'ElasticSearch index to query',
+        }),
+        required: true,
+      },
+      size: {
+        types: ['number'],
+        help: i18n.translate('data.search.eql.size.help', {
+          defaultMessage: 'ElasticSearch searchAPI size parameter',
+        }),
+        default: 10,
+      },
+      field: {
+        types: ['string'],
+        help: i18n.translate('data.search.eql.field.help', {
+          defaultMessage: 'List of fields to retrieve',
+        }),
+        multi: true,
+        required: false,
+      },
+    },
+    async fn(input, args, { inspectorAdapters, abortSignal, getKibanaRequest }) {
+      const { search, uiSettingsClient } = await getStartDependencies(getKibanaRequest);
+
+      const dsl: EqlSearchRequest['body'] = {
+        query: args.query,
+        size: args.size,
+        // @ts-ignore
+        fields: args.field,
+      };
```

Comment:
It is not a good practice to use `@ts-ignore` since it may suppress other errors. In this case, we can miss an error if we rename `args.field`.

It seems like we cannot easily augment the `body` property, so the best option will be to cast that explicitly.

```suggestion
      const dsl = {
        query: args.query,
        size: args.size,
        fields: args.field,
      } as unknown as Required<EqlSearchRequest>['body'];
```